GoogleForCreators/web-stories-wp
# Media: Video Segmentation Prototype

## Context

Automatically split videos into smaller parts.

Example: you upload a 2 minute video and we allow you to split it into 15 seconds long segments (so 8 parts in total), putting every segment on its own page.

For this prototype we can create a new feature flag and some simple button (and maybe a range slider to set the segmentation length) somewhere that triggers this action.



## Summary

Adds a segment video panel under styles when you select a video

<img width="250" alt="Screen Shot 2022-09-19 at 7 18 11 AM" src="https://user-images.githubusercontent.com/62242/191007131-89237b5c-c078-4f41-a1eb-50c77894515c.png">

Pressing segment will split the selected video into parts based on the number of seconds selected



## Relevant Technical Choices



## To-do



## User-facing changes



## Testing Instructions

QA Notes added some known issues for follow-up

https://github.com/GoogleForCreators/web-stories-wp/issues/12463
https://github.com/GoogleForCreators/web-stories-wp/issues/12464
https://github.com/GoogleForCreators/web-stories-wp/issues/12465

👉 Note: the video is segmented based on the videos [keyframes](https://en.wikipedia.org/wiki/Key_frame#:~:text=Whenever%20a%20drastic%20change%20to%20the%20image%20occurs%2C%20such%20as%20when%20switching%20from%20one%20camera%20shot%20to%20another%2C%20or%20at%20a%20scene%20change%2C%5B1%5D%20a%20key%20frame%20must%20be%20created) i.e. using a 30 sec segment length will not result in exactly 30 secs segments but rather the segments will be based on the closet keyframe.

> Note that splitting may not be accurate, unless you force the reference stream key-frames at the given time. See the introductory notice and the examples below.






This PR can be tested by following these steps:

- Enable the segment video feature
- upload a video to be segmented
- select the video from the canvas
- visit the style tab / video segment panel
- select the # of seconds to segment the video into
- press segment
- the video should get segmented + split onto multiple pages  



## Reviews

### Does this PR have a security-related impact?

No



### Does this PR change what data or activity we track or use?

No



### Does this PR have a legal-related impact?

No



## Checklist




---



Fixes #12164
